src/Storages/ColumnsDescription.cpp

```diff
@@ -59,6 +59,7 @@ bool ColumnDescription::operator==(const ColumnDescription & other) const
 
     return name == other.name
         && type->equals(*other.type)
+        && type->getName() == other.type->getName()
```

Comment:
Missing comments.

Reply:
The code is wrong because it applies not only to Replicated tables,
but we allow aggregate functions with compatible states for Distributed, Merge, and Buffer tables.